driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1473,6 +1476,36 @@ public Builder withCloudSecureConnectBundle(InputStream cloudConfigInputStream)
       return addCloudConfigToBuilder(cloudConfig);
     }
 
+    /**
+     * Disables advanced shard awareness. By default, this driver chooses local port while making a
+     * connection to node, to signal which shard it wants to connect to. This allows driver to
+     * estabilish connection pool faster, especially when there are multiple clients connecting
+     * concurrently. If this causes any issues, you can disable it using this method.
+     *
+     * @return this builder
+     */
+    public Builder withoutAdvancedShardAwareness() {
+      this.useAdvancedShardAwareness = false;
+      return this;
+    }
+
+    /**
+     * Sets local port range for use by advanced shard awareness. Driver will use ports from this
+     * range as local ports when connecting to cluster. If {@link #withoutAdvancedShardAwareness()}
+     * was called, then setting this range does not affect anything.
+     *
+     * @param low Lower bound of range, inclusive.
+     * @param high Upper bound of range, inclusive.
+     */
+    public Builder withLocalPortRange(int low, int high) {
+      if (low < 1 || 65535 < low || 1 < high || 65535 < high) {
```

Comment:
Nit: `low < high` and also some requirement that this range is sufficiently large (`low = 10000, high = 10001` not accepted).

Reply:
What minimum amount of ports in range would you propose? 1000?